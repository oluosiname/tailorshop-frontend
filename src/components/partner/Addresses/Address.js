import React from "react";
import Toggle from "../../toggle/Toggle";

const Address = ({ address, setPrimaryAddress, deleteAddress }) => {
  return (
    <React.Fragment>
      {address && (
        <div className="address">
          <div className="address__info">
            <div className="address__details">
              <p data-testid="address-address1"> {address.address1}</p>
              <div className="address__address2">
                <span data-testid="address-address2">{address.address2}</span>
              </div>
              <p>
                <span
                  style={{ letterSpacing: "1px" }}
                  data-testid="address-city"
                >
                  {address.city}
                </span>
                , <span data-testid="address-state">{address.state}</span>
              </p>
            </div>
            <div className="default">
              <span>Default</span>
              <div style={{ float: "right" }}>
                {" "}
                <Toggle
                  checked={address.primary}
                  onChange={() => setPrimaryAddress(address.uuid)}
                />
              </div>
            </div>
          </div>

          <div className="address__meta">
            <div className="edit">Edit</div>
            <div className="delete" onClick={() => deleteAddress(address.uuid)}>
              Delete
            </div>
          </div>
        </div>
      )}
    </React.Fragment>
  );
};

export default Address;
