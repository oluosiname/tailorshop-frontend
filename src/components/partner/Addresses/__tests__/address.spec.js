import React from "react";
import ReactDOM from "react-dom";
import Address from "../Address";
import { render, cleanup, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);
const address = {
  uuid: "uuid",
  address1: "Block 262",
  address2: "Flat 10",
  city: "Festac",
  state: "Lagos",
  primary: true,
};

const setPrimaryAddress = jest.fn((uuid) => {
  return;
});
const deleteAddress = jest.fn((uuid) => {
  return;
});

it("renderswithout crashing", () => {
  const div = document.createElement("div");

  ReactDOM.render(<Address />, div);
});

it("renders properly", () => {
  const { getByTestId } = render(
    <Address
      address={address}
      setPrimaryAddress={setPrimaryAddress}
      deleteAddress={deleteAddress}
    />
  );
  expect(getByTestId("address-address1")).toHaveTextContent("Block 262");
  expect(getByTestId("address-address2")).toHaveTextContent("Flat 10");
  expect(getByTestId("address-city")).toHaveTextContent("Festac");
  expect(getByTestId("address-state")).toHaveTextContent("Lagos");
});

it("matches snapshot", () => {
  const tree = renderer.create(<Address address={address} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("calls delete function", () => {
  const { getByTestId, getByText } = render(
    <Address
      address={address}
      setPrimaryAddress={setPrimaryAddress}
      deleteAddress={deleteAddress}
    />
  );
  fireEvent.click(getByText("Delete"));
  expect(deleteAddress).toHaveBeenCalledTimes(1);
  expect(deleteAddress).toHaveBeenCalledWith("uuid");

  fireEvent.click(getByTestId("toggle-primary-address"));
  expect(setPrimaryAddress).toHaveBeenCalledTimes(1);
  expect(setPrimaryAddress).toHaveBeenCalledWith("uuid");
});
