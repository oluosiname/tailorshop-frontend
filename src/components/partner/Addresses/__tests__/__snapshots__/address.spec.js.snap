// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="address"
>
  <div
    className="address__info"
  >
    <div
      className="address__details"
    >
      <p
        data-testid="address-address1"
      >
         
        Block 262
      </p>
      <div
        className="address__address2"
      >
        <span
          data-testid="address-address2"
        >
          Flat 10
        </span>
      </div>
      <p>
        <span
          data-testid="address-city"
          style={
            Object {
              "letterSpacing": "1px",
            }
          }
        >
          Festac
        </span>
        , 
        <span
          data-testid="address-state"
        >
          Lagos
        </span>
      </p>
    </div>
    <div
      className="default"
    >
      <span>
        Default
      </span>
      <div
        style={
          Object {
            "float": "right",
          }
        }
      >
         
        <label
          className="toggle-control"
          data-testid="toggle-primary-address"
        >
          <input
            checked={true}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="control"
          />
        </label>
      </div>
    </div>
  </div>
  <div
    className="address__meta"
  >
    <div
      className="edit"
    >
      Edit
    </div>
    <div
      className="delete"
      onClick={[Function]}
    >
      Delete
    </div>
  </div>
</div>
`;
